File: projekatZaIT355/src/main/java/com/metropolitan/models/Recenzija.java
package com.metropolitan.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Recenzija {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private int id;
    private String komentar;
    @Column(nullable = false)
    private int ocena;
    @ManyToOne
    @JoinColumn(name="korisnik_id")
    private Korisnik korisnik;
    @ManyToOne
    @JoinColumn(name="vozilo_id")
    @JsonIgnore
    private Vozilo vozilo;

    private boolean aktivna=false;

    private LocalDateTime vremeKreiranja;
}

File: projekat_za_it355_frontend/src/app/models/Recenzija/recenzija.ts
import { Korisnik } from "../Korisnik/korisnik";
import { Vozilo } from "../Vozilo/vozilo";

export class Recenzija {
    id:number;
    komentar: string;
    ocena:number;
    korisnik:Korisnik;
    vozilo:Vozilo;
    aktivna: boolean;
    vremeKreiranja: Date;
}
